tests/terratest/functions/waitForActiveCluster.go

```diff
@@ -0,0 +1,122 @@
+package functions
+
+import (
+	"testing"
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func WaitForActiveCluster(t *testing.T, client *rancher.Client, clusterID string, module string) {
+	if module == "aks" {
+	// Check for updating/waiting cluster state
+	err := wait.Poll(500*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
+		cluster, err := client.Management.Cluster.ByID(clusterID)
+		require.NoError(t, err)
+
+		if err != nil {
+			return false, err
+		}
+
+		if cluster.State == "updating" {
+			return true, nil
+		}
+
+		if cluster.State == "waiting" {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	require.NoError(t, err)
+
+
+	// Check for active cluster state
+	err = wait.Poll(500*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
+		cluster, err := client.Management.Cluster.ByID(clusterID)
+		require.NoError(t, err)
+
+		if err != nil {
+			return false, err
+		}
+
+		if cluster.State == "active" {
+			return true, nil
+		}
+
+		return false, nil
+	})
+	require.NoError(t, err)
+	time.Sleep(4 * time.Second)
+	}
+	
+	if module != "aks" {
+			// Check cluster state is active
+	err := wait.Poll(500*time.Millisecond, 30*time.Minute, func() (done bool, err error) {
```

Comment:
Nice work Josh, much better. Can indent this line though the formatting seems off and it's hard to read since it looks like this whole block of code is under the if statement

Reply:
Sure thing!